astropy/table/table.py

```diff
@@ -477,6 +477,12 @@ def __init__(self, data=None, masked=None, names=None, dtype=None,
                 # gets a key copy here if copy=False because later a direct object ref
                 # is used.
                 meta = data.meta if copy else data.meta.copy()
+            # If user-input meta is not None and data.meta also contains data
+            if meta is not None and data.meta:
+                if copy:
+                    meta = {**data.meta, **meta}
```

Comment:
If your goal is to concat dictionaries, this is the wrong syntax.

Reply:
Now using update() to concat